kotlin/src/main/java/com/squareup/moshi/KotlinJsonAdapter.kt

```diff
@@ -143,61 +144,61 @@ internal class KotlinJsonAdapter<T> private constructor(
       return if (value !== ABSENT_VALUE) value else null
     }
   }
+}
 
-  companion object {
-    @JvmField val FACTORY = Factory { type, annotations, moshi ->
-      if (!annotations.isEmpty()) return@Factory null
+object KotlinJsonAdapterFactory : JsonAdapter.Factory {
```

Comment:
It's accessed using `KotlinJsonAdapterFactory.INSTANCE` from Java in this PR.

Using `@file:JvmName("Something")` and naming it `FACTORY` would mean it's accessed like this from Java: `Something.FACTORY` and simply `FACTORY` from Kotlin. The latter seems to be worse than the downside of appearing as `KotlinJsonAdapterFactory.INSTANCE` from Java caused by this PR.

There's another option: Make `KotlinJsonAdapter` and thus the companion object public, but all of its non-inherited members `internal`. That may be error-prone though.